### PR TITLE
Add optional_permissions column to chrome_extensions table

### PR DESCRIPTION
**Pull request by rcipkins**
Chrome extensions have optional permissions which are defined in the extension manifest file along the required permissions. This PR will add an additional column to the chrome_extensions table to include optional_permissions.

### CODE REVIEW

File: osquery/tables/applications/browser_utils.cpp [hunk 1]
```diff
@@ -137,10 +138,37 @@ void genExtension(const std::string& uid,
     permission_list = boost::algorithm::join(perm_vector, ", ");
   }
 
+  // Fetch the optional permission array from the manifest file
+  std::string optional_permission_list;
```
Comment (theopolis) [inline osquery/tables/applications/browser_utils.cpp]: Could we collapse the `permissions` and `optional_permissions` into a function?

Comment (rcipkins) [reply osquery/tables/applications/browser_utils.cpp]: Yes definitely.

File: osquery/tables/applications/browser_utils.cpp [hunk 2]
```diff
@@ -71,6 +71,32 @@ Status getChromeProfileName(std::string& name, const fs::path& path) {
 }
 } // namespace
 
+const std::string genPermissions(const std::string& permissionTypeKey,
+                                 pt::ptree tree) {
```
Comment (theopolis) [inline osquery/tables/applications/browser_utils.cpp]: Does this work as a const reference? Eg., const pt::ptree& tree

Comment (rcipkins) [reply osquery/tables/applications/browser_utils.cpp]: It does. Should I change it to that?

Comment (theopolis) [reply osquery/tables/applications/browser_utils.cpp]: yes please! Passing the `tree` by reference will prevent an unneeded copy of the underlying data.

**Comment by theopolis** [comment]
The windows build error seems unrelated to this change.

**Comment by Smjert** [comment]
@rcipkins Please rebase on latest master, the Windows build failure will be fixed then.

**Comment by Smjert** [comment]
Closing and reopening to re-trigger the CI.

**Pull request merged**